ui/lib/kv/addon/components/page/list.js

```diff
@@ -87,6 +87,7 @@ export default class KvListPageComponent extends Component {
     evt.preventDefault();
     pathIsDirectory(this.secretPath)
       ? this.router.transitionTo('vault.cluster.secrets.backend.kv.list-directory', this.secretPath)
-      : this.router.transitionTo('vault.cluster.secrets.backend.kv.secret.index', this.secretPath);
+      : // navigate directly to the details because user has explicitly typed in a secret path
```

Comment:
Since this is a replacement for navigating in the list view, I disagree that we should go directly to details

Reply:
I think the diff may be confusing - this was the original workflow. It's when a user has permissions to read the secret but not list. What do you propose instead? The test was failing because navigating to the overview still works even though the secret isn't real. I don't think it makes sense to check whether or not the secret is real from the overview since we're trying to avoid make the request there.
<img width="838" alt="Screenshot 2024-08-28 at 9 26 25 AM" src="https://github.com/user-attachments/assets/9a32255d-2306-4ac4-9d33-58da8259cf44">

https://github.com/hashicorp/vault/blob/main/ui/lib/kv/addon/components/page/list.js#L86C1-L91

Reply:
Right, I would expect it to go to the overview page, same as if you clicked on one of the list items. I agree it doesn't make sense to check if the secret exists. The screenshot you shared seems fine, and as a future improvement on that page we could show some error state if the metadata request comes back as 404